sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchServiceAsyncClient.java

```diff
@@ -1286,4 +1302,13 @@ public String getEndpoint() {
             return monoError(logger, ex);
         }
     }
+
+    protected String getETagFromResource(Object resource, boolean onlyIfUnchanged) {
```

Comment:
Let's remove this method and keep ETag retrieval scoped to the individual call. Right now this is using reflection to retrieve a known field when using an `Index`, `Indexer`, etc.

Let's keep the logic simple and repetitive in the package private methods that call into the implementation layer by having them use `getETag` on the object they are working with.